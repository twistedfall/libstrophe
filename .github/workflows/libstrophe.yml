name: libstrophe
on:
  push:
    branches:
      - "*"
    tags_ignore:
      - "*"
  pull_request:
    branches:
      - "*"
jobs:
  ci:
    strategy:
      fail-fast: false
      matrix:
        libstrophe-version:
          - 0.9.3
          - 0.10.1
          - 0.11.0
          - 0.12.3
          - 0.13.1
          - 0.14.0
    runs-on: ubuntu-24.04
    env:
      LIBSTROPHE_VERSION: ${{ matrix.libstrophe-version }}
      SCCACHE_GHA_ENABLED: "true"
      RUSTC_WRAPPER: "sccache"
    steps:
      - uses: actions/checkout@v4
      - uses: mozilla-actions/sccache-action@v0.0.9
      - uses: dtolnay/rust-toolchain@stable
        with:
          components: clippy,rustfmt

      - run: ci/install.sh
        shell: bash

      - run: ci/script.sh
        shell: bash

  msrv:
    runs-on: ubuntu-24.04
    env:
      SCCACHE_GHA_ENABLED: "true"
      RUSTC_WRAPPER: "sccache"
    steps:
      - uses: actions/checkout@v4
      - uses: mozilla-actions/sccache-action@v0.0.9
      - uses: dtolnay/rust-toolchain@master
        with:
          toolchain: ${{ steps.metadata.outputs.msrv }}
          components: clippy,rustfmt

      - name: Get MSRV package metadata
        id: metadata
        run: cargo metadata --no-deps --format-version 1 | jq -r '"msrv="+(.packages[]|select(.name=="libstrophe").rust_version)' >> $GITHUB_OUTPUT

      - run: ci/checks.sh
        shell: bash
