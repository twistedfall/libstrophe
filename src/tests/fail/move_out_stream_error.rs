use libstrophe::{Connection, Context, StreamError};
use std::sync::Mutex;

fn main() {
	let stream_error = Mutex::new(Option::<StreamError>::None);
	let mut _handler = {
		move |_: &Context, _: &mut Connection, _: i32, _: i32, err: Option<StreamError>| {
			*stream_error.lock().unwrap() = err;
		}
	};
//		let conn = Connection::new(Context::new_with_null_logger());
//		conn.connect_client(None, None, &mut handler);
}
