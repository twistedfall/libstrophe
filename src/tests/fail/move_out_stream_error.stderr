error: borrowed data cannot be stored outside of its closure
 --> $DIR/move_out_stream_error.rs:8:36
  |
5 |     let mut stream_error = Mutex::new(Option::<StreamError>::None);
  |                                       --------------------------- cannot infer an appropriate lifetime...
6 |     let mut handler = {
  |         ----------- ...so that variable is valid at time of its declaration
7 |         move |_: &Context, _: &mut Connection, _: i32, _: i32, err: Option<StreamError>| {
  |         -------------------------------------------------------------------------------- borrowed data cannot outlive this closure
8 |             *stream_error.lock().unwrap() = err;
  |                                             ^^^ cannot be stored outside of its closure
